tools/lsp/properties.rs

```diff
@@ -339,30 +348,35 @@ mod tests {
         properties.iter().find(|p| p.name == name)
     }
 
-    fn properties_at_position_in_cache(
+    fn properties_at_position_in_cache<'a>(
         line: u32,
         character: u32,
-        dc: &mut crate::server_loop::DocumentCache,
-        url: &lsp_types::Url,
-    ) -> Option<Vec<PropertyInformation>> {
+        dc: &'a mut DocumentCache,
+        url: &'a lsp_types::Url,
```

Comment:
AFAICS the explicit lifetime is not needed (my build doesn't require it at least).

```suggestion
        dc: &mut DocumentCache,
        url: & lsp_types::Url,
```